lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -56,20 +62,32 @@ def names_and_raw_engine_configs
         end
       end
 
-      def include_paths
-        IncludePathsBuilder.new(exclude_paths, Array(@requested_paths)).build
+      def base_workspace
+        @base_workspace ||= Workspace::PathTree.new(".").tap do |path_tree|
+          path_tree.include_paths(@requested_paths) if @requested_paths.present?
+          path_tree.exclude_paths([".git"])
+          normalize_paths(@config.exclude_paths).each do |exclude|
+            path_tree.exclude_paths(Workspace::Exclusion.new(exclude).expand)
```

Comment:
I think it's a little leaky to let/expect callers to reach in for `PathTree` and `Exclusion` directly. WDYT about having `Workspace.create` (or something) to encapsulate the `PathTree` instantiation, then have `PathTree#exclude_paths` accept patterns and wrap them as `Exclusion`s internally?